File: CPP/FirstFileExample.cpp
#include<iostream>
#include<cstdio>
#include<fstream>

using namespace std;

int main()
{
	string str;
	ofstream  fout;		//fin is an object of ofstream class
	
	//-------WRITING CONTENT TO A FILE
	fout.open("FirstFile.txt");
	cout<<"Please type some text below:"<<endl;
	getline(cin, str);
	//gets(str);
	//fflush(stdin);
	fout<<str;
	cout<<"Entered information successfully saved in the file: FirstFile"<<endl;		
	fout.close();
	
	
	//------READING CONTENT FROM A FILE
	cout<<"To view the content of FirstFile press any key..."<<endl;
	fflush(stdin);
	getchar();
	ifstream fin;
	fin.open("FirstFile.txt");
	cout<<"FirstFile contains the following text:"<<endl;
	getline(fin,str);
	cout<<str<<endl;	
	fin.close();
	
	return 0;
}






/*
	The difference between cout and fout is that cout is pre-defined whereas we have to define/create
	fout by our own.
	cout/cin are connected to console whereas fout/fin are connected to files.
	
	
	Functionally they are same.
*/

File: VB/FirstFileExample.vb
imports System.io

Module FirstFile
	Sub Main()
	
		Dim str as String
		
	
		'-------WRITING CONTENT TO A FILE
		Dim sw as new StreamWriter("FirstFile.txt")		'sw is reference to object of StreamWriter class
		Console.WriteLine("Please type some text below:")
		str = Console.ReadLine()
		sw.WriteLine(str)
		Console.WriteLine("Entered information successfully saved in the file: FirstFile")
		sw.close()
	
	
		'------READING CONTENT FROM A FILE
		Console.WriteLine("To view the content of FirstFile press any key...")
		Console.ReadKey()
		Dim sr as new StreamReader("FirstFile.txt")
		Console.WriteLine("FirstFile contains the following text:")
		str = sr.ReadLine()
		Console.WriteLine(str)	
		sr.close()
	End Sub
End Module






	'I try to follow single paradigm as far as possible 
	'.Net exe will run through CLR

